lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -28,61 +41,151 @@ VAR
     fVoltageBuffer: ARRAY[1..1000] OF LREAL;
 
     {attribute 'pytmc' := '
-        pv: CALIB
+        pv: STC
         io: input
     '}
-    fCalibBase: LREAL;
+    fbTempSensor: FB_CC_TempSensor;
 
+
+    uFrequency: UINT;
+    // Used to get beam mode and rate
     {attribute 'pytmc' := '
-        pv: CALIB_BUFFER
-        io: input
+        pv: SXRAccTypeFromEpics
+        link: PHYS:UNDS:1:FACMODE.RVAL
+    '}
+    uSXR_Accelerator: UINT;
+    {attribute 'pytmc' := '
+        pv: SXR_NC_BeamRateFromEpics
+        link: EVNT:SYS0:1:NC_SOFTRATE
+    '}
+    uSXR_NC_Frequency: UINT;
```

Comment:
Small data typing thing, some of these links are DBR_DOUBLE in EPICS.
Do you know what EPICS does when the types don't match?
Hopefully it does something sensible like rounding.

Reply:
I assume it does round, but I changed the two EVNT:SYS0:1 links to be reals.